src/main/java/de/zalando/tip/zalenium/proxy/DockerSeleniumStarterRemoteProxy.java

```diff
@@ -134,20 +135,42 @@ public TestSession getNewSession(Map<String, Object> requestedCapability) {
 
     /*
         Starting a few containers (Firefox, Chrome), so they are ready when the tests come.
+        Executed in a thread so we don't wait for the containers to be created and the node
+        registration is not delayed.
     */
     @Override
     public void beforeRegistration() {
         readConfigurationFromEnvVariables();
-        if (getChromeContainersOnStartup() > 0 || getFirefoxContainersOnStartup() > 0) {
-            LOGGER.log(Level.INFO, LOGGING_PREFIX + "Setting up {0} Firefox nodes and {1} Chrome nodes ready to use.",
-                    new Object[]{getFirefoxContainersOnStartup(), getChromeContainersOnStartup()});
-            for (int i = 0; i < getChromeContainersOnStartup(); i++) {
-                startDockerSeleniumContainer(BrowserType.CHROME);
-            }
-            for (int i = 0; i < getFirefoxContainersOnStartup(); i++) {
-                startDockerSeleniumContainer(BrowserType.FIREFOX);
+        setupCompleted = false;
+        createStartupContainers();
+    }
+
+    private void createStartupContainers() {
+        Thread thread = new Thread() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 24 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1188)